### PR TITLE
[dtests[ Fix build issues with hipEnvVar*.cpp on windows

### PR DESCRIPTION
**Pull request by ansurya**
Platform specific implementation to use setenv(), popen, pclose system calls for linux and  _putenv_s, _popen, _pclose for windows. platform is detected with predefined macros.
Replaced getopt with clara based implementation which is platform independent.

### CODE REVIEW

**Comment by ansurya** [comment]
setenv(), popen, pclose system call requirements for windows platform is taken as is from PR #1281. Replaced getopt with clara based implementation.

**Comment by emankov** [comment]
LGTM. The only concern is about BSD 2-Clause license. Does it allow such manipulations with header files?

File: tests/src/hipEnvVar.cpp [hunk 1]
```diff
@@ -27,57 +27,61 @@ THE SOFTWARE.
```
Comment (gargrahul) [inline tests/src/hipEnvVar.cpp]: Is this test being run on CI?

**Comment by gargrahul** [comment]
These changes would need verification on Windows too.

File: tests/src/hipEnvVar.cpp (hunk 1 above)
Comment (ansurya) [reply tests/src/hipEnvVar.cpp]: This test is used by hipEnvVarDriver.cpp which is run on CI.

**Comment by ansurya** [comment]
These changes are verified by QA team on windows.

**Comment by mangupta** [comment]
> LGTM. The only concern is about BSD 2-Clause license. Does it allow such manipulations with header files?

The clara.hpp is already being used in HIP. See https://github.com/ROCm-Developer-Tools/HIP/blob/master/lpl_ca/clara/clara.hpp.

**Comment by emankov** [comment]
> The clara.hpp is already being used in HIP. See https://github.com/ROCm-Developer-Tools/HIP/blob/master/lpl_ca/clara/clara.hpp.

I see, but the question is in such a mixed usage of header files under the BSD 2-Clause license. Is it allowed, or the original header files should be used?

**Comment by mangupta** [comment]
> > The clara.hpp is already being used in HIP. See https://github.com/ROCm-Developer-Tools/HIP/blob/master/lpl_ca/clara/clara.hpp.
> 
> I see, but the question is in such a mixed usage of header files under the BSD 2-Clause license. Is it allowed, or the original header files should be used?

I am not sure which header are we modifying. We are using the single_include version of clara.hpp. We are also probably using an older version of the file. Most likely this one: https://github.com/catchorg/Clara/blob/6462131410c4c579ea60fea2db34845de88f468c/single_include/clara.hpp.

**Comment by emankov** [comment]
> I am not sure which header are we modifying. We are using the single_include version of clara.hpp. We are also probably using an older version of the file. Most likely this one: https://github.com/catchorg/Clara/blob/6462131410c4c579ea60fea2db34845de88f468c/single_include/clara.hpp.

I've been worrying about the following:
```
// ----------- end of #include from clara_textflow.hpp -----------
// ........... back in clara.hpp
```
As long as it is so in the original (single_include) version of Clara, FMPOV all is ok.

**Comment by ansurya** [comment]
> LGTM. Please write a couple of sentences about what this PR is done for.

Done.

**Pull request merged**